crates/subspace-networking/src/node.rs

```diff
@@ -68,49 +70,94 @@ impl PinnedDrop for TopicSubscription {
 
 #[derive(Debug, Error)]
 pub enum GetValueError {
-    /// Node runner was dropped, impossible to get value.
-    #[error("Node runner was dropped, impossible to get value")]
-    NodeRunnerDropped,
+    /// Failed to send command to the node runner
+    #[error("Failed to send command to the node runner: {0}")]
+    SendCommand(#[from] SendError),
+    /// Node runner didn't send result back
+    #[error("Node runner didn't send result back")]
+    ReceiveResult,
+}
+
+impl From<oneshot::Canceled> for GetValueError {
+    fn from(oneshot::Canceled: oneshot::Canceled) -> Self {
+        Self::ReceiveResult
+    }
 }
 
 #[derive(Debug, Error)]
 pub enum GetClosestPeersError {
-    /// Node runner was dropped, impossible to get closest peers.
-    #[error("Node runner was dropped, impossible to get closest peers")]
-    NodeRunnerDropped,
+    /// Failed to send command to the node runner
+    #[error("Failed to send command to the node runner: {0}")]
+    SendCommand(#[from] SendError),
+    /// Node runner didn't send result back
+    #[error("Node runner didn't send result back")]
+    ReceiveResult,
+}
+
+impl From<oneshot::Canceled> for GetClosestPeersError {
+    fn from(oneshot::Canceled: oneshot::Canceled) -> Self {
+        Self::ReceiveResult
+    }
 }
 
 #[derive(Debug, Error)]
 pub enum SubscribeError {
-    /// Node runner was dropped, impossible to subscribe.
-    #[error("Node runner was dropped, impossible to get value")]
-    NodeRunnerDropped,
+    /// Failed to send command to the node runner
+    #[error("Failed to send command to the node runner: {0}")]
+    SendCommand(#[from] SendError),
+    /// Node runner didn't send result back
+    #[error("Node runner didn't send result back")]
+    ReceiveResult,
     /// Failed to create subscription.
     #[error("Failed to create subscription: {0}")]
     Subscription(#[from] SubscriptionError),
 }
 
+impl From<oneshot::Canceled> for SubscribeError {
+    fn from(oneshot::Canceled: oneshot::Canceled) -> Self {
+        Self::ReceiveResult
+    }
+}
+
 #[derive(Debug, Error)]
 pub enum PublishError {
-    /// Node runner was dropped, impossible to publish.
-    #[error("Node runner was dropped, impossible to get value")]
-    NodeRunnerDropped,
+    /// Failed to send command to the node runner
+    #[error("Failed to send command to the node runner: {0}")]
+    SendCommand(#[from] SendError),
+    /// Node runner didn't send result back
+    #[error("Node runner didn't send result back")]
+    ReceiveResult,
```

Comment:
Hm... I liked `NodeRunnerDropped` better because it was more specific, can you elaborate on why was this change necessary?

Reply:
Hmm, didn't give it much thought. Makes sense, will revert it